src/vsr/grid.zig

```diff
@@ -141,11 +154,6 @@ pub fn GridType(comptime Storage: type) type {
         cache: Cache,
         /// Each entry in cache has a corresponding block.
         cache_blocks: []BlockPtr,
-        /// Each entry in cache has a corresponding bit.
-        /// This bit tracks whether the cache block is definitely valid relative to the current
-        /// checkpoint (matches what should be on disk). At the conclusion of state sync, all blocks
-        /// are marked as invalid. This enables additional validation of cache consistency.
-        cache_coherent: std.DynamicBitSetUnmanaged,
```

Comment:
Rather than tracking which cache entries are coherent ourselves (internal to the grid), each read must signal whether or not it can guarantee that its target is coherent.

Specifically, coherent reads are:
- opening the manifest log
- commit prefetch
- compaction

Other reads pass `coherent=false`:
- repair requests (via the network)
- scrubbing (in a follow-up PR). Scrubbing is technically coherent at the time that the read is submitted, but marking it as incoherent allows the read to not block checkpoints. (Which simplifies things, since this means that reads can never block checkpoints.)